### DISCUSSION TITLE
Is the scan channel hopping sequence SEQUENTIAL ou RANDOM?

### DISCUSSION THREAD

**Discussion by rpelicia** [h2zero/NimBLE-Arduino#603] [Q&A] [answered]
Hi!
In a passive scan, each window only listens to one of the three available advertisement channels.
What happens if my scan duration (1000 ms) fits only two windows (500 ms each)? Will only channels 37 and 38 ever be scanned, or will all three channels (37, 38 and 39) have the same share of attention as long as I start enough scans?

Pseudocode:
```
        pBLEScan->setActiveScan(false);
        pBLEScan->setInterval(500);
        pBLEScan->setWindow(500);
        while (1) {
                pBLEScan->start(1, false);
                // DO OTHER THINGS ...
        }
```
Thanks!!!

**Comment by h2zero** [comment, marked as answer]
Great question, I don't really have an answer for that. I would guess that it was sequential, would need to look in the controller code or possibly it would be in the specifications.

**Reply by rpelicia** [reply]
I am doing a few statistical tests to see if I learn something.

**Reply by rpelicia** [reply]
**Quick statistical investigation parameters:**
          100 "starts" for each test.
          RSSI = 50-70 dBm
          Beacon advertisement interval: 480 ms.
          NimBLE-Arduino: 
          
          
          pBLEScan->setActiveScan(false);
          pBLEScan->setInterval(500);
          pBLEScan->setWindow(500);
          while (1) {
                  pBLEScan->start(1, false);
                  // DO OTHER THINGS ...
          }
          

**Results:**
    Advertising only on CH 37: 33% of scans detected the beacon
    Advertising only on CH 38: 45% of scans detected the beacon
    Advertising only on CH 39: 33% of scans detected the beacon
    Advertising on all channels: 96% of scans detected the beacon

At a first glance, the scan channel-hopping sequence appears to be random, perhaps with some favoritism for channel 38. Further investigation is warranted.
